Sources/PrimerSDK/Classes/Data Models/AsyncPaymentMethodTokenizationViewModel.swift

```diff
@@ -0,0 +1,394 @@
+//
+//  PaymentMethodViewModel.swift
+//  PrimerSDK
+//
+//  Created by Evangelos Pittas on 7/10/21.
+//
+
+import Foundation
+
+typealias TokenizationCompletion = ((PaymentMethodToken?, Error?) -> Void)
+
+internal protocol PaymentMethodTokenizationViewModelProtocol {
+    func startTokenizationFlow()
+    var completion: TokenizationCompletion? { get set }
+}
+
+class AsyncPaymentMethodTokenizationViewModel {
+    
+    internal private(set) var config: PaymentMethodConfig
+    private var webViewController: WebViewController?
+    internal var position: Int!
+    var tokenizationCompletion: TokenizationCompletion!
+    private let theme: PrimerThemeProtocol = DependencyContainer.resolve()
+    internal private(set) var paymentMethod: PaymentMethodToken?
+    
+    deinit {
+        log(logLevel: .debug, message: "🧨 deinit: \(self) \(Unmanaged.passUnretained(self).toOpaque())")
+    }
+    
+    init(config: PaymentMethodConfig) {
+        self.config = config
+    }
+    
+    @objc
+    func buttonTapped() {
+        switch config.type {
+        case .hoolah,
```

Comment:
when it comes to switching between all the payment methods, maybe we could add enum methods to reduce the number of touch points for every new payment method? We did a similar change to Android.